Remotion/Web/Core/UI/Controls/ControlItemCollection.cs

```diff
@@ -98,17 +99,17 @@ private void OnCollectionChanged (CollectionChangeEventArgs e)
         CollectionChanged (this, e);
     }
 
-    protected virtual void ValidateNewValue (object value)
+    protected virtual void ValidateNewValue ([NotNull]object? value)
```

Comment:
Hmm... mainly, I don't want to mark `value` as nullable or we should annotate `value` as not-null after the method was called. The other places that deal with `value` (`OnInsert`, `OnInsertComplete`) don't need to have `value` nullable after ValidateNewValue ensured that it can't be null.

Reply:
This is represented by the current state:
```csharp
protected virtual void ValidateNewValue ([NotNull]object? value)
```
Value could be `null`, but is ensured to not be `null` via the post condition `NotNull`. After the invocation of `ValidateNewValue`, `value` is not null.